src/com/pablo/utils/Inventario.java

```diff
@@ -0,0 +1,161 @@
+package com.pablo.utils;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Inventario {
+	private double iva;
+	private double tasa;
+	private Scanner scan = new Scanner(System.in);
+
+	public void agregarInventario(ArrayList<Producto> inventarioProductos,String descripcion,String codigo, int cantidad, double precio) {
+		inventarioProductos.add(new Producto(descripcion, codigo, cantidad, precio));
+	}
+
+	public void listadoInvetario(ArrayList<Producto> inventarioProductos) {
+		for (int i = 0; i < inventarioProductos.size(); i++) {
+			System.out.println("-------------------------------");
+			System.out.println("Nombre " + inventarioProductos.get(i).getDescripcion());
+			System.out.println("Codigo " + inventarioProductos.get(i).getCodigo());
+			System.out.println("Cantidad " + inventarioProductos.get(i).getCantidad());
+			System.out.println("Precio " + inventarioProductos.get(i).getPrecio());
+			System.out.println("-------------------------------");
+		}
+	}
+
+	public void listadoProductosMasMenosCostoso(ArrayList<Producto> inventarioProductos, double masMenosCostoso) {
+		for (int i = 0; i < inventarioProductos.size(); i++) {
+			if (masMenosCostoso == inventarioProductos.get(i).getPrecio()) {
+				System.out.println("Nombre " + inventarioProductos.get(i).getDescripcion());
+			}
+		}
+	}
+
+	public void listadoProductosMasMenosCant(ArrayList<Producto> inventarioProductos, int masMenosCant) {
+		for (int i = 0; i < inventarioProductos.size(); i++) {
+			if (masMenosCant == inventarioProductos.get(i).getCantidad()) {
+				System.out.println("Nombre " + inventarioProductos.get(i).getDescripcion());
+			}
+		}
+	}
+
+	public void modificarInventario(ArrayList<Producto> inventarioProductos, String codigo) {
+		for (int i = 0; i < inventarioProductos.size(); i++) {
```

Comment:
lo mismo, para no manejar for, puedes usar el for in, 

for(Producto producto : productos){
    // mi logica
}

o usar la API Collections con la clase Optional., te recuerdo que esto es un MUST para trabajar con Java.

Optional<Producto> producto = productos.stream().filter( producto -> producto.equals(codigo)).firstFirst();
//con esto puedes valir si filter encontro el producto en el arrayList.
if(producto.isEmpty())